### ISSUE TITLE
Update Documentation and Example Instructions - REST API

### ISSUE THREAD

**Issue by LGRmac** [exogee-technology/graphweaver#967]
**Describe the bug**
The documentation for connecting to a REST API is incomplete. Following the documentation to completion does not lead to a successful connection between GraphWeaver and a REST API. (https://graphweaver.com/docs/connect-to-a-rest-api)
Additionally, the instructions on the REST API example page, do not contain enough instructions to allow for a correct implementation of the code.

From the Code Example (https://github.com/exogee-technology/graphweaver/tree/main/src/examples/rest)
- It is never explained why MySQL is necessary in this example, when the idea is to connect a REST API.
- The Authentication is never explained. It would appear we are authenticating with the MySQL database, but what does this have to do with the REST example?
- Instructions to start the project are incomplete as the error "You need to build the cli package first" is always thrown - regardless of the pnpm command given (ie. pnpm build, pnpm i etc.)
- Following the project initiation instructions from the GraphWeaver documentation, and copying the files from the GitHub example also does not work. The documentation states that GraphWeaver will automatically look for custom page files; however, an error is thrown saying the files do not exist. (https://graphweaver.com/docs/add-custom-pages-to-the-admin-ui)

From the documentation (https://graphweaver.com/docs/connect-to-a-rest-api)
- A filename and location for the final code snippet of the page, is not given. Hence this code is orphaned and is never run.
- This documentation is not sufficient to get a working example - the server will run correctly but no calls are made to the API - nor is it explained how that works. The admin page is blank and no data is fetched.

**To Reproduce**
https://github.com/exogee-technology/graphweaver/tree/main/src/examples/rest
https://graphweaver.com/docs/connect-to-a-rest-api

- Custom Project with GitHub Example
1. run `npx graphweaver@latest init`
2. select MySQL and REST
3. Copy all files from github
4. pnpm i
5. pnmp start

**Expected behavior**
Successful connection to a REST API - with clear instructions on how to make API calls.

**Actual behavior**
GraphWeaver server is started with none of the expected functionality - and no indication of any errors or issues in the console.
GraphWeaver server fails to start.
GraphWeaver server starts but fails to load due to not finding custom page files.

**Screenshots or Screencasts**
- Build Error
![image](https://github.com/user-attachments/assets/66406f5a-3948-40fa-96df-0f45fc264438)
- Missing Page Files
![image](https://github.com/user-attachments/assets/f84df1b1-6567-41cf-a14b-4acc8ba2da5c)

**Desktop (please complete the following information):**

- Windows 10 Enterprise v22H2
- Node v20.16.0
- pnpm v9.6.0

**Comment by thekevinbrown** [comment]
Hey @LGRmac, thanks for giving this a go and letting us know about this! I'll respond point by point.

* Yup, you're absolutely right. The REST example started out as a simple demo of how to join REST to MySQL. When we built auth, that was added to this example, and it now demonstrates very complex auth scenarios. It's way more complex than it needs to be for just simple REST which is what you're more interested in. I've gone ahead and created a simple REST example that just does REST. [You can see it here](https://github.com/exogee-technology/graphweaver/tree/chore/add-simpler-rest-example/src/examples/rest) while I work on getting it merged into main. Once merged it'll just be `examples/rest` in the repo.
* In the example we're actually authing with users in REST and credentials in the database. This shows that even users can come from external to Graphweaver. We're trying to demo that unlike just about every other GraphQL API solution, we don't even care where users come from. We don't need to own your data in Graphweaver, we just need to be able to make decisions getting the data from _somewhere_.
* You probably don't need to do this at all (see below), but if you want to get the examples running in the monorepo you'll need to run `pnpm build` in the `src` directory. We will add this step to the instructions for the examples as all of them will require this step.
* If you're not interested in contributing to Graphweaver yet you don't need our monorepo checked out at all. I've created a standalone version you can just download, install dependencies, and run. It's here: https://github.com/exogee-technology/graphweaver-issue-967-standalone-example

On the docs, you're right, they're out of date, thank you for flagging. I've added a task to fix that up, but I wanted to make sure I unblocked you in the meantime. Let me know if that gets you moving!

**Comment by thekevinbrown** [comment]
Just an update that the new simpler REST example is merged into `main`.

**Comment by LGRmac** [comment]
@thekevinbrown Thanks mate, you've gone above and beyond here! The standalone example is really helpful - I didn't have the whole repo checked out, nor had I run `pnpm build` in the `src` directory,  so that would explain a few things on my end.  Appreciate the help - everything is much clearer already!

**Comment by thekevinbrown** [comment]
Awesome, glad to hear that was helpful! I'll leave this open until we've finished actioning all of the things we need to here:

1. ~~Create simple REST example, rename `rest` to `rest-with-auth`~~
2. ~~Share standalone version of simple REST example so monorepo complexities are avoided~~
3. Update Connect to a REST API docs and ensure they're accurate
4. Add `pnpm build` step to example readme files for clarity

**Comment by thekevinbrown** [comment]
Hey there, I believe this is all done at this point, so I'll close this off. If you have any problems with the new REST adapter, just let us know!
